controllers/gateway/resource.go

```diff
@@ -29,23 +31,49 @@ import (
 
 // buildServiceResource builds a new service that exposes gateway.
 func (ctx *gatewayContext) buildServiceResource() (*corev1.Service, error) {
```

Comment:
Shall we build Service resource? Or shall we only build a ClusterIP service? Follow-up is needed.